src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsprobationestateapi/integration/IntegrationTestBase.kt

```diff
@@ -48,11 +47,9 @@ abstract class IntegrationTestBase {
     if (!regionRepository.existsById(region.code)) {
       regionRepository.save(region)
     }
-    probationDeliveryUnitRepository.existsById(probationDeliveryUnit.code)
-      .filter { !it }
-      .flatMap { probationDeliveryUnitRepository.save(probationDeliveryUnit) }
-      .then(
-        teamRepository.save(team)
-      ).block()!!
+    if (!probationDeliveryUnitRepository.existsById(probationDeliveryUnit.code)) {
+      probationDeliveryUnitRepository.save(probationDeliveryUnit)
+    }
+    teamRepository.save(team)
```

Comment:
A minor point but why check that the region and PDU exist but not the team?

Reply:
I think it was intention, a team shouldn't exist when you're trying to set it up